test/library/client.go

```diff
@@ -429,18 +415,19 @@ func CreateTestLDAPIdentityProvider(t *testing.T, spec idpv1alpha1.LDAPIdentityP
 
 	// Wait for the LDAPIdentityProvider to enter the expected phase (or time out).
 	var result *idpv1alpha1.LDAPIdentityProvider
-	require.Eventuallyf(t, func() bool {
-		var err error
-		result, err = upstreams.Get(ctx, created.Name, metav1.GetOptions{})
-		if err != nil {
-			t.Logf("error while getting LDAPIdentityProvider %s/%s: %s", created.Namespace, created.Name, err.Error())
-			return false
-		}
-		return result.Status.Phase == expectedPhase
-	},
+	RequireEventuallyf(t,
+		func(requireEventually *require.Assertions) {
+			var err error
+			result, err = upstreams.Get(ctx, created.Name, metav1.GetOptions{})
+			requireEventually.NoErrorf(err, "error while getting LDAPIdentityProvider %s/%s", created.Namespace, created.Name)
+			requireEventually.Equal(expectedPhase, result.Status.Phase)
+		},
 		2*time.Minute, // it takes 1 minute for a failed LDAP TLS connection test to timeout before it tries using StartTLS, so wait longer than that
 		1*time.Second,
-		"expected the LDAPIdentityProvider to go into phase %s, LDAPIdentityProvider was: %s", expectedPhase, Sdump(result))
+		"expected the LDAPIdentityProvider to go into phase %s, LDAPIdentityProvider was: %s",
+		expectedPhase,
+		Sdump(result),
```

Comment:
This is not a comment on your change, but rather a comment on the original line of code from before your change, which I think I just spotted that I made a mistake when I originally wrote it...

Don't all the arguments of `RequireEventuallyf()` need to be evaluated before `RequireEventuallyf()` can be called? In other words, isn't `Sdump(result)` evaluated before `RequireEventuallyf()` is called, so the value of `result` would alway be `nil` in the failure output? 🤔

Reply:
Thanks for noticing this. Fixed in the latest change.